<?xml version="1.0"?>
<robot name = "openDog">

	<material name="blue">
        <color rgba="0 0 0.8 1"/>
    </material>

    <material name="red">
        <color rgba="0.8 0 0 1"/>
    </material>

    <material name="green">
        <color rgba="0 0.8 0 1"/>
    </material>

    <material name="grey">
        <color rgba="0.75 0.75 0.75 1"/>
    </material>

    <material name="white">
        <color rgba="1.0 1.0 1.0 1"/>
    </material>

    <material name="black">
        <color rgba="0 0 0 1"/>
    </material>


    <link name = "openDog_body">
    	<visual>
    		<origin xyz = "0 0 0.7" rpy = "0 0 0"/>
    		<geometry>
    			<box size = "0.31 0.881 0.195"/>
    		</geometry>
    		<material name = "blue"/>
    	</visual>
    </link>

    <link name = "hip_tube_1">
    	<visual>
    		<origin xyz = "0.0 0.0 0.0" rpy = "0 1.5708 0"/>
    		<geometry>
    			<cylinder length = "0.165" radius = "0.05715"/>
    		</geometry>
    		<material name = "green"/>
    	</visual>
    </link>

    <link name = "hip_tube_2">
    	<visual>
    		<origin xyz = "0.0 0.0 0.0" rpy = "0 1.5708 0"/>
    		<geometry>
    			<cylinder length = "0.165" radius = "0.05715"/>
    		</geometry>
    		<material name = "green"/>
    	</visual>
    </link>

    <link name = "hip_tube_3">
    	<visual>
    		<origin xyz = "0.0 0.0 0.0" rpy = "0 1.5708 0"/>
    		<geometry>
    			<cylinder length = "0.165" radius = "0.05715"/>
    		</geometry>
    		<material name = "green"/>
    	</visual>
    </link>

    <link name = "hip_tube_4">
    	<visual>
    		<origin xyz = "0.0 0.0 0.0" rpy = "0 1.5708 0"/>
    		<geometry>
    			<cylinder length = "0.165" radius = "0.05715"/>
    		</geometry>
    		<material name = "green"/>
    	</visual>
    </link>

	<joint name="hip_joint_1" type="fixed">
    	<parent link="openDog_body"/>
    	<child link="hip_tube_1"/>
    	<origin xyz="-0.226 0.2825 0.7" rpy="0 0 0"/>
	</joint>

	<joint name="hip_joint_2" type="fixed">
    	<parent link="openDog_body"/>
    	<child link="hip_tube_2"/>
    	<origin xyz="0.226 0.2825 0.7" rpy="0 0 0"/>
	</joint>

	<joint name="hip_joint_3" type="fixed">
    	<parent link="openDog_body"/>
    	<child link="hip_tube_3"/>
    	<origin xyz="-0.226 -0.2825 0.7" rpy="0 0 0"/>
	</joint>

	<joint name="hip_joint_4" type="fixed">
    	<parent link="openDog_body"/>
    	<child link="hip_tube_4"/>
    	<origin xyz="0.226 -0.2825 0.7" rpy="0 0 0"/>
	</joint>



	<joint name="femur_joint1" type="revolute">
    	<parent link="hip_tube_1"/>
    	<child link="upper_extrusion_leg1"/>
    	<origin xyz="-0.05 0 0" rpy="1.5708 -3.14 1.5708"/>
        <limit lower="0" upper="0.44" effort="0.1" velocity="0.005"/>
        <axis xyz="0 0 1"/>
	</joint>


	<joint name="femur_joint2" type="revolute">
    	<parent link="hip_tube_2"/>
    	<child link="upper_extrusion_leg2"/>
    	<origin xyz="0.05 0 0" rpy="1.5708 -3.14 1.5708"/>
        <limit lower="0" upper="0.44" effort="0.1" velocity="0.005"/>
        <axis xyz="0 0 1"/>
	</joint>

	<joint name="femur_joint3" type="revolute">
    	<parent link="hip_tube_3"/>
    	<child link="upper_extrusion_leg3"/>
    	<origin xyz="-0.05 0 0" rpy="1.5708 -3.14 1.5708"/>
        <limit lower="0" upper="0.44" effort="0.1" velocity="0.005"/>
        <axis xyz="0 0 1"/>
	</joint>

	<joint name="femur_joint4" type="revolute">
    	<parent link="hip_tube_4"/>
    	<child link="upper_extrusion_leg4"/>
    	<origin xyz="0.05 0 0" rpy="1.5708 -3.14 1.5708"/>
        <limit lower="0" upper="0.44" effort="0.1" velocity="0.005"/>
        <axis xyz="0 0 1"/>
	</joint>

	<joint name = "knee_joint_leg1" type = "revolute">
		<parent link = "upper_extrusion_leg1"/>
		<child link = "lower_extrusion_leg1"/>
		<origin xyz="0 0.39 0" rpy="0 0 0"/>
		
		<limit effort="30" velocity="1.0" lower="-2.2" upper="0.7" />
		<axis xyz="0 0 1"/>
		
	</joint>

	<joint name = "knee_joint_leg2" type = "revolute">
		<parent link = "upper_extrusion_leg2"/>
		<child link = "lower_extrusion_leg2"/>
		<origin xyz="0 0.39 0" rpy="0 0 0"/>
		<limit effort="30" velocity="1.0" lower="-2.2" upper="0.7" />
		<axis xyz="0 0 1"/>
		
	</joint>

	<joint name = "knee_joint_leg3" type = "revolute">
		<parent link = "upper_extrusion_leg3"/>
		<child link = "lower_extrusion_leg3"/>
		<origin xyz="0 0.39 0" rpy="0 0 0"/>
		<limit effort="30" velocity="1.0" lower="-2.2" upper="0.7" />
		<axis xyz="0 0 1"/>
	</joint>

	<joint name = "knee_joint_leg4" type = "revolute">
		<parent link = "upper_extrusion_leg4"/>
		<child link = "lower_extrusion_leg4"/>
		<origin xyz="0 0.39 0" rpy="0 0 0 "/>
		<limit effort="30" velocity="1.0" lower="-2.2" upper="0.7" />
		<axis xyz="0 1 0"/>
	</joint>

	<link name = "lower_extrusion_leg1">
		<visual>
			<origin xyz = "0.0 0.155 0.0" rpy = "1.5708 3.14 0"/>
			<geometry>
			 	<box size = "0.02 0.04 0.31"/>
			</geometry>
			<material name = "grey"/>
		</visual>
	</link>

	<link name = "lower_extrusion_leg2">
		<visual>
			<origin xyz = "0.0 0.155 0.0" rpy = "1.5708 3.14 0"/>
			<geometry>
			 	<box size = "0.02 0.04 0.31"/>
			</geometry>
			<material name = "grey"/>
		</visual>
	</link>

	<link name = "lower_extrusion_leg3">
		<visual>
			<origin xyz = "0.0 0.155 0.0" rpy = "1.5708 3.14 0"/>
			<geometry>
			 	<box size = "0.02 0.04 0.31"/>
			</geometry>
			<material name = "grey"/>
		</visual>
	</link>

	<link name = "lower_extrusion_leg4">
		<visual>
			<origin xyz = "0.0 0.155 0.0" rpy = "1.5708 3.14 0"/>
			<geometry>
			 	<box size = "0.02 0.04 0.31"/>
			</geometry>
			<material name = "grey"/>
		</visual>
	</link>
	

	<link name = "upper_extrusion_leg1">
		<visual>
			<origin xyz = "0.0 0.195 0.0" rpy = "1.5708 0 0"/>
			<geometry>
				<box size = "0.02 0.06 0.39"/>
			</geometry>
			<material name = "red"/>
		</visual>
	</link>

	<link name = "upper_extrusion_leg2">
		<visual>
			<origin xyz = "0.0 0.195 0.0" rpy = "1.5708 0 0"/>
			<geometry>
				<box size = "0.02 0.06 0.39"/>
			</geometry>
			<material name = "red"/>
		</visual>
	</link>

	<link name = "upper_extrusion_leg3">
		<visual>
			<origin xyz = "0.0 0.195 0.0" rpy = "1.5708 0 0"/>
			<geometry>
				<box size = "0.02 0.06 0.39"/>
			</geometry>
			<material name = "red"/>
		</visual>
	</link>


	<link name = "upper_extrusion_leg4">
		<visual>
			<origin xyz = "0.0 0.195 0.0" rpy = "1.5708 0 0"/>
			<geometry>
				<box size = "0.02 0.06 0.39"/>
			</geometry>
			<material name = "red"/>
		</visual>
	</link>

	
</robot>
